airflow/jobs/scheduler_job_runner.py

```diff
@@ -1308,9 +1308,19 @@ def _create_dag_runs_dataset_triggered(
                     DatasetDagRunQueue.target_dag_id == dag_run.dag_id
                 ).delete()
 
-    def _should_update_dag_next_dagruns(self, dag, dag_model: DagModel, total_active_runs: int) -> bool:
+    def _should_update_dag_next_dagruns(
+        self, dag: DAG, dag_model: DagModel, session: Session, total_active_runs: int | None = None
+    ) -> bool:
```

Comment:
```suggestion
    def _should_update_dag_next_dagruns(
        self,
        dag: DAG,
        dag_model: DagModel,
        total_active_runs: int | None = None,
        *,
        session: Session,
    ) -> bool:
```

Also adjust the callers to match the signature. Having `session` in the middle of the arguments is a bit messy to me.